src/widgets/preprocessing_widget.py

```diff
@@ -1,13 +1,100 @@
+        self.plot_picker_submit.setText("Plot")
+        self.plot_picker_submit.setGeometry(QRect(10, 120, 50, 23))
+        self.plot_picker_submit.clicked.connect(lambda: self.plot_data(self.column_select_box.currentText(),
+                                                                       self.plot_select_box.currentText()))
+
+        # estimation group
+        self.estimate_group = QGroupBox(self.frame)
+        self.estimate_group.setTitle("Estimate missing values")
+        self.estimate_group.setGeometry(QRect(330, 30, 270, 120))
+
+        self.estimate_group_todo = QLabel(self.estimate_group)
+        self.estimate_group_todo.setText("Future")
+        self.estimate_group_todo.setGeometry(QRect(10, 40, 100, 23))
+
+        # automatic reduction group
+        self.auto_reduction_group = QGroupBox(self.frame)
+        self.auto_reduction_group.setTitle("Reduce dimensions automatically")
+        self.auto_reduction_group.setGeometry(QRect(630, 30, 270, 120))
+
+        self.auto_reduction_todo = QLabel(self.auto_reduction_group)
+        self.auto_reduction_todo.setText("Future")
+        self.auto_reduction_todo.setGeometry(QRect(10, 40, 100, 23))
+
+        # plot stats window
+        self.plot_widget = QGroupBox(self.frame)
+        self.plot_widget.setTitle("Result")
+        self.plot_widget.setGeometry(QRect(30, 200, 420, 350))
+
+        self.plot_layout = QVBoxLayout()
+        self.plot_widget.setLayout(self.plot_layout)
+
+        # column rejection group
+        self.columns_group = QGroupBox(self.frame)
+        self.columns_group.setTitle("Columns")
+        self.columns_group.setGeometry(QRect(480, 200, 420, 350))
+        self.columns_grid = QGridLayout()
+        self.columns_group.setLayout(self.columns_grid)
+        self.set_columns_grid()
+
+    def get_data(self):
+        """ check column names every time coming to that frame (potential changes) """
+        self.parent().unfold(1)
+        self.column_select_box.clear()
+        self.column_select_box.addItems(self.engine.get_columns())
+        self.set_columns_grid()
+
+    def plot_data(self, column_name, plot_type):
+        self._clear_plot()
+        plot_box = self.engine.create_plot(column_name, plot_type)
+        self.plot_layout.addWidget(plot_box)
+
+    def _clear_plot(self):
+        for i in reversed(range(self.plot_layout.count())):
+            self.plot_layout.itemAt(i).widget().setParent(None)
+
+    def set_columns_grid(self):
+        columns = self.engine.get_columns()
+        col = max(len(columns) // 11 + 1, 2)
+        rows = (len(columns) - 1) // col + 1
+        self.columns_group.setFixedHeight(min(rows * 60, 350))
+        positions = [(i, j) for i in range(rows) for j in range(col)]
+        for name, position in zip(columns, positions):
+            checkbox = QCheckBox(name)
+            checkbox.setChecked(True)
+            self.columns_grid.addWidget(checkbox, *position)
```

Comment:
Has the choice of columns some impact to the next step in app? Do you save this choice?